include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1239,7 +1239,7 @@ __pattern_stable_partition(_ExecutionPolicy&& __exec, _Iterator __first, _Iterat
     auto true_count = copy_result.first - __true_result;
 
     //TODO: optimize copy back if possible (inplace, decrease number of submits)
-    __pattern_walk2</*_IsSync=*/::std::false_type>(
+    __pattern_walk2(
         __par_backend_hetero::make_wrapped_policy<copy_back_wrapper>(__exec),
```

Comment:
Here it seems that these two `__pattern_walk2` calls are independent of each other (don't require a specific ordering), but both do need to be waited on before returning from `__pattern_stable_partition`. 

Instead of enforcing ordering here, could we make them both async calls and then wait on the policy's queue here?

Reply:
I agree  here...
Yes, we have to call wait for two events here. But we still don't have such oneDPL API. 
Usage of SYCL API directly is not allowed on this oneDPL code layer.

Reply:
Yes, you are correct.  I agree that similar to the others, this requires a rewrite / refactor to improve by returning some event from `__pattern_walk2` rather than the simple true / false async we have.

Reply:
What about if we rewrite this code like
`C++
    auto future1 = __pattern_walk2_async(__tag, __par_backend_hetero::make_wrapped_policy<copy_back_wrapper>(__exec),
                                         __true_result, copy_result.first, __first,
                                         __brick_move<__hetero_tag<_BackendTag>, _ExecutionPolicy>{});

    auto future2 = __pattern_walk2_async(
        __tag, __par_backend_hetero::make_wrapped_policy<copy_back_wrapper2>(::std::forward<_ExecutionPolicy>(__exec)),
        __false_result, copy_result.second, __first + true_count,
        __brick_move<__hetero_tag<_BackendTag>, _ExecutionPolicy>{});

    future1.wait();
    future2.wait();
```